### PR TITLE
{lib}[foss/2021a] igraph v0.9.4

### PR DESCRIPTION
**Pull request by jfgrimm**
(created using `eb --new-pr`)

### CODE REVIEW

**Comment by verdurin** [comment]
Test report by @verdurin
**SUCCESS**
Build succeeded for 2 out of 2 (1 easyconfigs in total)
nuc.lan - Linux Fedora 33, x86_64, Intel(R) Core(TM) i7-8650U CPU @ 1.90GHz, Python 3.9.5
See https://gist.github.com/d05120aa668e89e6ea6404ad022153b6 for a full test report.

File: easybuild/easyconfigs/i/igraph/igraph-0.9.4-foss-2021a.eb [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+dependencies = [
+    ('GLPK', '5.0'),
+    ('libxml2', '2.9.10'),
+    ('zlib', '1.2.11'),
+    ('SuiteSparse', '5.10.1', '-METIS-5.1.0'),
+    ('arpack-ng', '3.8.0'),
+]
+
+# Build static and shared libraries
+configopts = ["-DBUILD_SHARED_LIBS=OFF", "-DBUILD_SHARED_LIBS=ON"]
```
Comment (verdurin) [inline easybuild/easyconfigs/i/igraph/igraph-0.9.4-foss-2021a.eb:36]: @jfgrimm could you explain this? Looks weird to disable and enable the same option...

Also, was it deliberate that you removed the previous `configopts = "--with-external-blas --with-external-lapack --with-external-glpk"`?

File: easybuild/easyconfigs/i/igraph/igraph-0.9.4-foss-2021a.eb [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    'mv %(installdir)s/include/igraph/* %(installdir)s/include/ && rmdir %(installdir)s/include/igraph ',
+]
+
+_includes = ['include/igraph%s.h' % x for x in ['', '_blas', '_constants', '_lapack', '_types', '_version']]
+_libs = ['lib/libigraph.%s' % x for x in ['a', SHLIB_EXT]]
+
+sanity_check_paths = {
+    'files': _includes + _libs,
+    'dirs': [],
+}
```
Comment (verdurin) [inline easybuild/easyconfigs/i/igraph/igraph-0.9.4-foss-2021a.eb:49]: Is there a `sanity_check_command` we can run? Doesn't matter if not.

File: easybuild/easyconfigs/i/igraph/igraph-0.9.4-foss-2021a.eb (hunk 1 above)
Comment (jfgrimm) [reply easybuild/easyconfigs/i/igraph/igraph-0.9.4-foss-2021a.eb:36]: The old configopts are no longer supported (or at least, some of them; didn't exclusively test all). It looks like it correctly picks up and links the EB libraries nonetheless.

Comment (jfgrimm) [reply easybuild/easyconfigs/i/igraph/igraph-0.9.4-foss-2021a.eb:36]: The build scripts they use don't support building both static and shared libs at the same time. I guess that could be patched, but building twice (once with static, once shared) works without needing a patch

Comment (verdurin) [reply easybuild/easyconfigs/i/igraph/igraph-0.9.4-foss-2021a.eb:36]: Ah, yes - I saw the two iterations. Thanks for confirming.

File: easybuild/easyconfigs/i/igraph/igraph-0.9.4-foss-2021a.eb (hunk 2 above)
Comment (verdurin) [reply easybuild/easyconfigs/i/igraph/igraph-0.9.4-foss-2021a.eb:49]: I guess not...

**Comment by verdurin** [comment]
@boegelbot please test @ generoso

File: easybuild/easyconfigs/i/igraph/igraph-0.9.4-foss-2021a.eb (hunk 2 above)
Comment (jfgrimm) [reply easybuild/easyconfigs/i/igraph/igraph-0.9.4-foss-2021a.eb:49]: Not sure, I'm on leave and don't have my laptop with me so can't check

Comment (verdurin) [reply easybuild/easyconfigs/i/igraph/igraph-0.9.4-foss-2021a.eb:49]: Yes, not important - just nice to add more of these checks as we go.

**Review by verdurin — APPROVED** [review]
Looks fine.

**Comment by verdurin** [comment]
Going in, thanks @jfgrimm!

**Comment by verdurin** [comment]
Test report by @verdurin
**SUCCESS**
Build succeeded for 2 out of 2 (1 easyconfigs in total)
easybuild-c7.novalocal - Linux centos linux 7.9.2009, x86_64, Intel Xeon Processor (Skylake, IBRS), Python 3.6.8
See https://gist.github.com/08527c67ec77257c44e6a5d8318b2cb7 for a full test report.

**Pull request merged**